### PR TITLE
[2.0.x] Specify StorageClassName for PVC

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

**Review by oraNod — APPROVED** [review]
LGTM @dmvolod I added the docs commit to this PR as well.

**Comment by dmvolod** [comment]
We catch #801 in this test also for 2.0.x

File: documentation/asciidoc/topics/ref_container_storage.adoc [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+{cacheservice} and {datagridservice} nodes. You can configure storage resources
+for {datagridservice} nodes but not {cacheservice} nodes.
 
 [source,options="nowrap",subs=attributes+]
 ----
 include::yaml/cr_container_storage.yaml[]
 ----
 
 <1> Configures the storage size for {datagridservice} nodes.
+<2> Defines whether storage is ephemeral or permanent. Set the value to `true` to use ephemeral storage, which means all data in storage is deleted when clusters shut down or restart. The default value is `false`, which means storage is permanent.
```
Comment (rigazilla) [inline documentation/asciidoc/topics/ref_container_storage.adoc:13]: Do we have ephemeral storage in 2.0.x? Isn't it only on master?

File: documentation/asciidoc/topics/ref_datagridservice_resources.adoc [hunk 2]
```diff
@@ -11,8 +11,10 @@ spec:
   replicas: 6 <2>
   service:
     type: DataGrid <3>
-    container:
-      storage: 2Gi <4>
+    container: <4>
+      storage: 2Gi
+      ephemeralStorage: false
```
Comment (rigazilla) [inline documentation/asciidoc/topics/ref_datagridservice_resources.adoc:16]: same

File: documentation/asciidoc/topics/yaml/cr_container_storage.yaml [hunk 3]
```diff
@@ -4,3 +4,5 @@ spec:
     type: DataGrid
     container:
       storage: 2Gi <1>
+      ephemeralStorage: false <2>
```
Comment (rigazilla) [inline documentation/asciidoc/topics/yaml/cr_container_storage.yaml:7]: same

File: documentation/asciidoc/topics/ref_container_storage.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/asciidoc/topics/ref_container_storage.adoc]: whoops. I've removed ephemeral storage from this PR. nice catch @rigazilla

File: documentation/asciidoc/topics/ref_datagridservice_resources.adoc (hunk 2 above)
Comment (oraNod) [reply documentation/asciidoc/topics/ref_datagridservice_resources.adoc]: removed

File: documentation/asciidoc/topics/yaml/cr_container_storage.yaml (hunk 3 above)
Comment (oraNod) [reply documentation/asciidoc/topics/yaml/cr_container_storage.yaml]: removed

**Comment by ryanemerson** [comment]
@dmvolod Can you rebase

**Comment by dmvolod** [comment]
Re-based, but need review from @oraNod as a bit confusing with removed doc in current 2.0.x

**Comment by oraNod** [comment]
@dmvolod I removed reference to the `ephemeral` storage type that is not available on 2.0.x.

master: https://github.com/infinispan/infinispan-operator/blob/4bf99b09b87555f8061f1cc32f6b0c09d1e84483/pkg/apis/infinispan/v1/infinispan_types.go#L48

2.0.x: https://github.com/infinispan/infinispan-operator/blob/08bc46912a8daac07d63697bdbd1a3b156e276c0/pkg/apis/infinispan/v1/infinispan_types.go#L47 

This PR adds `StorageClassName` but not `EphemeralStorage` right?

**Comment by dmvolod** [comment]
> This PR adds `StorageClassName` but not `EphemeralStorage` right?

Yes, right

**Review by oraNod — APPROVED** [review]
Thanks @dmvolod I did notice that some yaml changed after a rebase and needed to add storage class name to the DG service reference.

Otherwise docs LGTM and do not mention type ephemeral.

**Comment by dmvolod** [comment]
Thanks, @oraNod 
Commits squashed

**Pull request merged**